play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -141,7 +142,23 @@ private[jackson] class JsValueDeserializer(factory: TypeFactory, klass: Class[_]
 
     val (maybeValue, nextContext) = (jp.getCurrentToken.id(): @switch) match {
 
-      case JsonTokenId.ID_NUMBER_INT | JsonTokenId.ID_NUMBER_FLOAT => (Some(JsNumber(jp.getDecimalValue)), parserContext)
+      case JsonTokenId.ID_NUMBER_INT | JsonTokenId.ID_NUMBER_FLOAT => {
+        val inputText = jp.getText
+        val inputLength = inputText.length
+
+        // There is a limit of how large the numbers can be since parsing extremely
+        // large numbers (think thousand of digits) and operating on the parsed values
+        // can potentially cause a DDoS.
+        if (inputLength > JacksonJson.BigDecimalLimits.DigitsLimit) throw new IllegalArgumentException("Number is larger than supported")
+
+        // Must create the BigDecimal with a MathContext that is consistent with the limits used.
+        val bigDecimal = BigDecimal(inputText, JacksonJson.BigDecimalLimits.DefaultMathContext)
```

Comment:
This is one of the crucial changes: using `MathContext.DECIMAL128` imposes a precision of 34 and then numbers requiring a higher precision will be truncated. But, a precision of 34 is good enough for most real cases.